### PR TITLE
Set up separate secrets for our staging server.

### PR DESCRIPTION
**Pull request by jfly**
Previously, we shared secrets between our staging and prod server, which
is a bad idea for all sorts of reasons. One of the biggest issues we've
run into is when our staging server started syncing our google group
mailing lists based off of the contents of the staging database
(https://github.com/thewca/worldcubeassociation.org/issues/4039). We
shouldn't have even been running that job on our staging server, but
even if someone *did* try to run it on our staging server, the server
should not have had the api key necessary to talk to our gsuite account!

This change is a step in the right direction, but isn't perfect:

- There's still a single chef secret that the staging server must have
  access to. With that secret, it would still be possible to decrypt the
  prod secrets and access prod stuff.
- We're using the same S3 IAM credentials for staging and prod right
  now. Ideally we'd create separate IAM users with only the permissions
  they need to access the staging/prod S3 buckets.

### CODE REVIEW

File: chef/site-cookbooks/wca/libraries/wca_helper.rb [hunk 1]
```diff
@@ -12,11 +12,14 @@ def self.get_username_and_repo_root(recipe)
   end
 
   def self.get_secrets(recipe)
-    if recipe.node.chef_environment.start_with?("development")
+    if recipe.node.chef_environment == "development"
```
Comment (jfly) [inline chef/site-cookbooks/wca/libraries/wca_helper.rb]: We used to have multiple kinds of development environments, but that is long gone, so I took this opportunity to simplify things.

File: chef/data_bags/secrets/production.json [hunk 2]
```diff
@@ -12,7 +12,13 @@
...
     "cipher": "aes-256-cbc"
   },
-  "mysql_password": {
+  "mysql_prod_password": {
+    "encrypted_data": "swX/3e9ap30MOHpSP0Ax4gfNgKVfP3kI8Aw/eQnsJd94OYrRJVl8XRrTGxHq\n0pYdMCM13dcJlLO/fsmCYARX7Q==\n",
+    "iv": "RQufC1hLQBb6XwhTlD0TfQ==\n",
+    "version": 1,
+    "cipher": "aes-256-cbc"
+  },
+  "mysql_staging_password": {
```
Comment (jonatanklosko) [inline chef/data_bags/secrets/production.json]: Is this intentional? Looks like we still only use `mysql_password` in `chef/site-cookbooks/wca/recipes/default.rb`.

File: chef/site-cookbooks/wca/recipes/default.rb [hunk 3]
```diff
@@ -183,8 +185,11 @@
 end
 
 # Use HTTPS in non development mode
-https = !node.chef_environment.start_with?("development")
-server_name = { "production" => "www.worldcubeassociation.org", "staging" => "staging.worldcubeassociation.org", "development" => "" }[node.chef_environment]
+https = (node.chef_environment != "development")
+server_name = {
+  "production" => "www.worldcubeassociation.org",
+  "staging" => "staging.worldcubeassociation.org", "development" => ""
```
Comment (jonatanklosko) [inline chef/site-cookbooks/wca/recipes/default.rb]: Nit: development deserves its own line ^^

File: chef/data_bags/secrets/production.json (hunk 2 above)
Comment (jfly) [reply chef/data_bags/secrets/production.json]: Oops! Thank you, that was definitely not intentional.

File: chef/site-cookbooks/wca/recipes/default.rb (hunk 3 above)
Comment (jfly) [reply chef/site-cookbooks/wca/recipes/default.rb]: Ah, good call, fixed!

**Comment by jfly** [comment]
Merged! I'm going to try spinning up a fresh staging server now.

**Comment by jfly** [comment]
That did not quite work =( See https://github.com/thewca/worldcubeassociation.org/pull/4680.

**Pull request merged**